app/Http/Controllers/Api/ExperienceSkillsController.php

```diff
@@ -48,4 +49,70 @@ public function destroy(ExperienceSkill $experienceSkill)
     {
         $experienceSkill->delete();
     }
+
+    public function batchStore(BatchStoreExperienceSkill $request)
+    {
+        $newExperienceSkills = $request->validated();
+        $response = [];
+
+        foreach ($newExperienceSkills as $newExperienceSkill) {
```

Comment:
@Khristophor had some good suggestions on how to deal with batch requests. One thing would be to wrap this for loop (and similarly in the other batch requests) in a [DB::transaction](https://laravel.com/docs/8.x/database#database-transactions) to ensure we don't end up with a partially-failed-partially-successful request.